README.md

```diff
@@ -1562,6 +1564,42 @@ _You can enable the following settings in Xcode by running [this script](resourc
   planets.first { $0.isGasGiant }
   ```
 
+  </details>
+
+* <a id='prefer-for-loop-over-forEach'></a>(<a href='#prefer-for-loop-over-forEach'>link</a>) **Prefer using for loops over the functional `forEach { ... }` method**, unless using `forEach` as the last element in a functional chain. [![SwiftFormat: forLoop](https://img.shields.io/badge/SwiftFormat-forLoop-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#forLoop)
+
+  <details>
+
+  #### Why?
+  For loops are more idiomatic than the `forEach` method, and are typically familiar to all developers who have experience with C-family languages. 
+
+  For loops are also more expressive than the `forEach` method. For loops support the `return`, `continue`, and `break` control flow keywords, while `forEach` only supports `return` (which has the same behavior as `continue` in a for loop).
+  
+  ```swift
+  // WRONG
+  planets.forEach { planet in
+    planet.terraform()
+  }
+
+  // WRONG
+  planets.forEach {
+    $0.terraform()
+  }
+
+  // RIGHT
+  for planet in planets {
+    planet.terraform()
+  }
+
+  // ALSO FINE, since forEach is useful when paired with other functional methods in a chain.
+  planets
+    .filter { !$0.isGasGiant }
+    .map { PlanetTerraformer(planet: $0) }
+    .forEach { $0.terraform() }
```

Comment:
By default the SwiftFormat rule avoids applying the rule in two cases:

1. If the subject of the `forEach` call uses trailing closure syntax, like `planets.filter { !$0.isGasGiant }.forEach { print($0) }` (handled [here](https://github.com/nicklockwood/SwiftFormat/pull/1490/files#diff-c7bca807450072a460ec31802d5f845f47ba9d85c040e437f3d8c3ebb56b2cd2R8039) by virtue of `}` not being handled in one of the switch cases -- should realistically be called out more explicitly).

We exclude this case because converting this to a for loop causes a new warning to be emitted:

```swift
// warning: warning: trailing closure in this context is confusable with the body of the statement; pass as a parenthesized argument to silence this warning
for planet in planets.filter { !$0.isGasGiant } {
  print(planet)
}
```

2. If the subject of the `forEach` call includes line breaks (handled [here](https://github.com/nicklockwood/SwiftFormat/pull/1490/files#diff-c7bca807450072a460ec31802d5f845f47ba9d85c040e437f3d8c3ebb56b2cd2R8077) with an explicit check). For example, it would be weird to convert:

```swift
planets
  .filter({ !$0.isGasGiant })
  .map({ PlanetTerraformer($0) })
  .forEach({ $0.terraform() })
```

to a for loop formatted something like

```swift
for planet in planets
  .filter({ !$0.isGasGiant })
  .map({ PlanetTerraformer($0) })
{
  print(planet)
}
```

A typical use case of chained functional operations hits both of these rules.

Reply:
> If the subject of the forEach call uses trailing closure syntax, like planets.filter { !$0.isGasGiant }.forEach { print($0) } (handled [here](https://github.com/nicklockwood/SwiftFormat/pull/1490/files#diff-c7bca807450072a460ec31802d5f845f47ba9d85c040e437f3d8c3ebb56b2cd2R8039) by virtue of } not being handled in one of the switch cases -- should realistically be called out more explicitly).

I improved how this is handled in the code here: https://github.com/nicklockwood/SwiftFormat/pull/1494/commits/4d92ba2b6b6f209aa3cad25f95eb0e4a29bf1618

Reply:
Thank you for explaining @calda and for that follow up!